From 1d14427cb973095787d921dbd4f925c770c09ee8 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 6 Oct 2016 19:42:55 -0700
Subject: [PATCH 106/203] bcm2835-camera: fix compilation warning/werror

| /a/builder/mnt/build/tmp-glibc/work-shared/raspberrypi3/kernel-source/drivers/media/platform/bcm2835/bcm2835-camera.c:656:17: error: initialization from incompatible pointer type [-Werror=incompatible-pointer-types
]
|   .queue_setup = queue_setup,
|                  ^~~~~~~~~~~
| /a/builder/mnt/build/tmp-glibc/work-shared/raspberrypi3/kernel-source/drivers/media/platform/bcm2835/bcm28
35-camera.c:656:17: note: (near initialization for 'bm2835_mmal_video_qops.queue_setup')

use struct device* instead of void*

Signed-off-by: Khem Raj <raj.khem@gmail.com>
[Xulin: Original patch taken from
https://github.com/raspberrypi/linux.git branch rpi-4.8.y]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/media/platform/bcm2835/bcm2835-camera.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/media/platform/bcm2835/bcm2835-camera.c b/drivers/media/platform/bcm2835/bcm2835-camera.c
index 70f4db2..cb5bab6 100644
--- a/drivers/media/platform/bcm2835/bcm2835-camera.c
+++ b/drivers/media/platform/bcm2835/bcm2835-camera.c
@@ -244,7 +244,7 @@ static struct mmal_fmt *get_format(struct v4l2_format *f)
 
 static int queue_setup(struct vb2_queue *vq,
 		       unsigned int *nbuffers, unsigned int *nplanes,
-		       unsigned int sizes[], void *alloc_ctxs[])
+		       unsigned int sizes[], struct device *alloc_ctxs[])
 {
 	struct bm2835_mmal_dev *dev = vb2_get_drv_priv(vq);
 	unsigned long size;
-- 
1.7.5.4

